plugin/api.go

```diff
@@ -480,6 +480,36 @@ type API interface {
 	//
 	// Minimum server version: 5.7
 	SendMail(to, subject, htmlBody string) *model.AppError
+
+	// CreateBot creates the given bot and corresponding user.
+	//
+	// Minimum server version: 5.8
+	CreateBot(bot *model.Bot) (*model.Bot, *model.AppError)
+
+	// PatchBot applies the given patch to the bot and corresponding user.
+	//
+	// Minimum server version: 5.8
+	PatchBot(userId string, botPatch *model.BotPatch) (*model.Bot, *model.AppError)
+
+	// GetBot returns the given bot.
+	//
+	// Minimum server version: 5.8
+	GetBot(userId string, includeDeleted bool) (*model.Bot, *model.AppError)
+
+	// GetBots returns the requested page of bots.
+	//
+	// Minimum server version: 5.8
+	GetBots(page, perPage int, creatorId string, includeDeleted bool) ([]*model.Bot, *model.AppError)
```

Comment:
page, perPage should be final arguments since that's what we do everywhere else.

Reply:
I'm going to pivot to a `model.BotGetOptions` to make this cleaner, given we're already at four parameters anyway.